src/App.tsx

```diff
@@ -34,62 +34,88 @@ export function getReorderedGoods(
+  }
 
-  return visibleGoods;
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+  render() {
+    const goods = getReorderedGoods(
+      goodsFromServer,
+      this.state,
+    );
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info${this.state.sortType !== 1 ? ' is-light' : ''}`}
+            onClick={() => this.setState({ sortType: 1 })}
+          >
+            Sort alphabetically
+          </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+          <button
+            type="button"
+            className={`button is-success${this.state.sortType !== 2 ? ' is-light' : ''}`}
+            onClick={() => this.setState({ sortType: 2 })}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={`button is-warning${!this.state.isReversed ? ' is-light' : ''}`}
+            onClick={() => this.setState(prev => ({
+              isReversed: !prev.isReversed,
+            }))}
+          >
+            Reverse
+          </button>
+
+          {this.state.sortType !== 0 || this.state.isReversed
+            ? (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={() => this.setState({
+                  isReversed: false,
+                  sortType: 0,
+                })}
+              >
+                Reset
+              </button>
+            )
+            : <></>}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {goods.map((item => (
+              <li key={item} data-cy="Good">{item}</li>
```

Comment:
hmm, setting an entire object as a key?
does the item have an `id`? If it is, i'd rather go with `item.id`
```suggestion
              <li key={item.id} data-cy="Good">{item}</li>
```

Reply:
No this is just an array of strings

```js
export const goodsFromServer = [
  'Dumplings',
  'Carrot',
  'Eggs',
  'Ice cream',
  'Apple',
  'Bread',
  'Fish',
  'Honey',
  'Jam',
  'Garlic',
];
```